vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/gemini_server.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_ENDPOINT_GEMINI_GEMINI_SERVER_H_
+#define BRAVE_VENDOR_BAT_NATIVE_LEDGER_SRC_BAT_LEDGER_INTERNAL_ENDPOINT_GEMINI_GEMINI_SERVER_H_
+
+#include <memory>
+
+#include "bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.h"
+#include "bat/ledger/internal/endpoint/gemini/post_balance/post_balance_gemini.h"
+#include "bat/ledger/internal/endpoint/gemini/post_oauth/post_oauth_gemini.h"
+#include "bat/ledger/internal/endpoint/gemini/post_transaction/post_transaction_gemini.h"
+#include "bat/ledger/ledger.h"
+
+namespace ledger {
+class LedgerImpl;
+
+namespace endpoint {
+
+class GeminiServer {
+ public:
+  explicit GeminiServer(LedgerImpl* ledger);
```

Comment:
`LedgerImpl*` is injected throughout the codebase to call back to `LedgerImpl` which then calls out to the `LedgerClient`. As a temporary measure to clean up the code check out the solution from the ads library, see `vendor/bat-native-ads/src/bat/ads/internal/ads_client_helper.h`.

Reply:
This convention exists throughout the custodian code, I think fixing this as a follow-up will reduce inconsistency between different custodians- https://github.com/brave/brave-browser/issues/16876